### PR TITLE
Better support for service accounts in helm and other helm improvements

### PR DESCRIPTION
**Pull request by mwylde**
This PR makes a small improvement to the helm chart to improve the process of configuring service accounts for k8s deployments. This was in service of making it easier to configure Arroyo to run on GKE and talk to GCS, which typically is done via a service account (as described [here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#using_from_your_code)). Specifically, we previously would only forward the configured service account to the workers if `serviceAccount.create` was `true` in the helm chart, whereas now we will do so if `serviceAccount.name` is non-empty.

This PR also adds the ability to configure the deployments to use a pre-existing configmap as environment variables, which makes it easier to set arbitrary configs across our services.

Finally, it cleans up some leftover config from the old API gRPC service ports, as now only the HTTP port is needed.

### CODE REVIEW

File: arroyo-controller/src/schedulers/kubernetes.rs [hunk 1]
```diff
@@ -83,15 +89,11 @@ impl KubernetesScheduler {
...
         }
     }
-}
-
-#[async_trait]
-impl Scheduler for KubernetesScheduler {
-    async fn start_workers(&self, req: StartPipelineReq) -> Result<(), SchedulerError> {
-        let api: Api<ReplicaSet> = Api::default_namespaced(self.client.clone());
 
+    fn make_replicaset(&self, req: StartPipelineReq) -> ReplicaSet {
```
Comment (jacksonrnewhouse) [inline arroyo-controller/src/schedulers/kubernetes.rs:96]: `make_replica_set`?

File: k8s/arroyo/templates/NOTES.txt [hunk 2]
```diff
@@ -6,11 +6,11 @@ Once the release is fully deployed, you can navigate to the web ui by running
 
   $ open "http://$(kubectl get service/{{ .Release.Name }}-api -o jsonpath='{.spec.clusterIP}')"
 
-(note this might take a minute or so if you are also deploying Postgres)
+(note this might take a few minutes so if you are also deploying Postgres)
```
Comment (jacksonrnewhouse) [inline k8s/arroyo/templates/NOTES.txt:9]: wording. should be "note that this might take a few minutes if you are also deploying Postgres"

**Review by jacksonrnewhouse — COMMENTED** [review]
Just a couple typos, looks good otherwise.

File: arroyo-controller/src/schedulers/kubernetes.rs (hunk 1 above)
Comment (mwylde) [reply arroyo-controller/src/schedulers/kubernetes.rs:96]: no, replicaset is one word

**Review by jacksonrnewhouse — APPROVED** [review]

**Pull request merged**